source/Dgraph/Channels/CloudChannel.cs

```diff
@@ -16,30 +16,63 @@
 
 using Grpc.Core;
 using Grpc.Net.Client;
-using System.Threading.Tasks;
 
 namespace Dgraph
 {
     public static class DgraphCloudChannel
     {
+        private const string CloudPort = "443";
+
+        /// <summary>
+        /// Create a new TLS connection to a Dgraph Cloud backend.
+        /// </summary>
+        /// <returns>A new instance of <see cref="GrpcChannel"/></returns>
+        /// <exception cref="ArgumentException"></exception>
         public static GrpcChannel Create(string address, string apiKey)
         {
-            var credentials = CallCredentials.FromInterceptor((context, metadata) =>
+            if (string.IsNullOrWhiteSpace(address))
+            {
+                throw new ArgumentException($"Invalid address to Dgraph Cloud: {address}");
+            }
+            if (string.IsNullOrWhiteSpace(apiKey))
+            {
+                throw new ArgumentException("Invalid api key for Dgraph Cloud");
+            }
+
+            string grpcUri = address;
+            if (!grpcUri.StartsWith("http"))
+            {
+                grpcUri = $"https://{grpcUri}";
+            }
+            if (Uri.TryCreate(grpcUri, UriKind.Absolute, out Uri u) && u.Host.Contains("."))
             {
-                if (!string.IsNullOrEmpty(apiKey))
+                if (u.Host.Contains(".grpc."))
                 {
-                    metadata.Add("DG-Auth", apiKey);
+                    grpcUri = $"https://{u.Host}:{CloudPort}";
                 }
+                else
+                {
+                    var uriParts = u.Host.Split(".", 2);
+                    grpcUri = $"https://{uriParts[0]}.grpc.{uriParts[1]}:{CloudPort}";
+                }
+            }
+            else
+            {
+                throw new ArgumentException($"Invalid address to Dgraph Cloud: {address}");
+            }
+
+            var credentials = CallCredentials.FromInterceptor((context, metadata) =>
+            {
+                metadata.Add("DG-Auth", apiKey);
```

Comment:
I am not sure we still use this. @rderbier or @kevinmingtarja do you know?

Reply:
@sachyco can you try with "authorization" instead and see if that works?

Reply:
Done. Ran some tests against my cloud instance, works as expected.